NetworkingSampleApp/NetworkingSampleApp/Scenes/Download/DownloadProgressViewModel.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  DownloadRowViewModel.swift
+//  NetworkingSampleApp
+//
+//  Created by Matej Molnár on 07.03.2023.
+//
+
+import SwiftUI
+import Networking
+
+final class DownloadProgressViewModel: ObservableObject {
+    private let task: URLSessionTask
+    
+    @Published var state: DownloadProgressState = .init()
+    
+    init(task: URLSessionTask) {
+        self.task = task
+    }
+    
+    func onAppear() {
+        Task {
```

Comment:
you might be able to make this function async and call it on the `.task` instead of onAppear